ml_genn/ml_genn/readouts/avg_var.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+from .readout import Readout
+from ..utils.model import NeuronModel
+
+from copy import deepcopy
+
+
+class AvgVar(Readout):
+    def add_readout_logic(self, model: NeuronModel, **kwargs):
+        self.output_var_name = model.output_var_name
+
+        if "var_name_types" not in model.model:
+            raise RuntimeError("AvgVar readout can only be used "
+                               "with models with state variables")
+        if self.output_var_name is None:
+            raise RuntimeError("AvgVar readout requires that models "
+                               "specify an output variable name")
+
+        # Find output variable
+        try:
+            output_var = [v for v in model.model["var_name_types"]
+                          if v[0] == self.output_var_name]
+        except StopIteration:
+            raise RuntimeError(f"Model does not variable "
```

Comment:
model does not *have* ...?

Reply:
Good spot - all these error messages were badly mangled and the actual code to find the output variable also didn't work - that code will never raise ``StopIteration``